src/main/java/net/glowstone/net/handler/play/inv/WindowClickHandler.java

```diff
@@ -42,7 +42,7 @@ private boolean process(final GlowPlayer player, final WindowClickMessage messag
         final InventoryView view = player.getOpenInventory();
         final GlowInventory top = (GlowInventory) view.getTopInventory();
         final GlowInventory bottom = (GlowInventory) view.getBottomInventory();
-        final ItemStack slotItem = view.getItem(viewSlot);
+        final ItemStack slotItem = message.getItem();
```

Comment:
the view.getItem(viewSlot) was null, caused the action in WindowLogic something or other to return NOTHING - nerfing most crafting actions/clicks. maybe my change masks a more serious issue elsewhere?